pkg/lib/authn/otp/form.go

```diff
@@ -52,6 +52,15 @@ func (f Form) GenerateCode(cfg *config.TestModeConfig, featureCfg *config.TestMo
 		}
 		return c.Generate()
 	case secretcode.LinkOTPSecretCodeType:
+		if cfg.Email.Enabled {
+			if r, ok := cfg.Email.MatchTarget(target); ok {
+				fixedCode := r.FixedCode
+				if fixedCode == "" {
+					fixedCode = c.Generate()
+				}
```

Comment:
We should follow this order

1. If the rule has fixed code, use it
2. Otherwise, if feature config has enabled deterministic code, use it (c.GenerateDeterministic(userID))
3. Otherwise, use random code.